### ISSUE TITLE
Ground Truth mask label is used

### ISSUE THREAD

**Issue by bio-mlhui** [MICV-yonsei/EAGLE#4]
I saw that the ground truth label is using when training:

```python
feats, feats_kk, code, code_kk = self.net(img)
...
detached_code_kk = torch.clone(code_kk.detach())
...
linear_logits = self.linear_probe(detached_code_kk)
linear_logits = F.interpolate(linear_logits, label.shape[-2:], mode='bilinear', align_corners=False)
linear_logits = linear_logits.permute(0, 2, 3, 1).reshape(-1, self.n_classes)
linear_loss = self.linear_probe_loss_fn(linear_logits[mask], flat_label[mask]).mean()
loss += linear_loss
...
manual_backward(loss)
```
In my opinion, alghough `detached_code_kk` does not requires gradient. The `linear_logits` requires gradient and will generate a new torch graph. The `linear_loss` will backward gradient to the `linear_probe`. Does this mean the ground truth pixel-wise mask label is used when training? 

I am very confused. It would be very kind for you to teach me where I am wrong. Thanks a lot!
